.github/workflows/deploy_terraform.yml

```diff
@@ -14,24 +14,67 @@ env:
 jobs:
   pre_job:
     name: Set Build Environment
-    concurrency:
-      group: ${{ github.workflow }}-${{ needs.pre_job.outputs.env_name }}
+    concurrency: 
+      group: ${{ github.workflow }}-${{ github.run_id }}
```

Comment:
I am not sure run_id will match and cancel as expected.

Reply:
This will work:
group: ${{ github.workflow }}-${{ needs.pre_job.outputs.env_name }}